cfme/tests/infrastructure/test_host.py

```diff
@@ -312,3 +312,39 @@ def _finalize():
             actual_state, power_state
         ),
     )
+
+
+@test_requirements.general_ui
+def test_infrastructure_hosts_refresh_multi(
+    appliance, setup_provider, provider
+):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Infra
+        caseimportance: low
+        initialEstimate: 1/3h
+        testSteps:
+            1. Navigate to the Compute > Infrastructure > Hosts view.
+            2. Select at least 2 hosts (more if possible) by checking the box in upper left of
+               quadicons.
+            3. Click "Refresh Relationships and Power States" under the Configuration
+               dropdowm, and then click "OK" when prompted.
+        expectedResults:
+            1. Hosts view is displayed.
+            2.
+            3. "Refresh initiated for X Hosts from the CFME Database" is displayed in green
+               banner where "X" is the number of selected hosts. Properties for each host are
+               refreshed.
+    """
+    if len(provider.data.get('hosts', {})) < 2:
+        pytest.skip('not enough hosts to run test')
+    hosts_view = navigate_to(appliance.collections.hosts, "All")
+
+    assert hosts_view.is_displayed
+    [h.check() for h in hosts_view.entities.get_all()]
+    num_checked_hosts = len(hosts_view.entities.get_all())
```

Comment:
I would store this as a variable at the start of the test to reduce redundancy, it should always be the same as the number of hosts correct? (Unless a provider happened to have an inane no. of hosts on it)